python/ray/experimental/iter.py

```diff
@@ -0,0 +1,556 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import TypeVar, Generic, Iterable, List, Callable, Any
+
+import ray
+
+# The type of an iterator element.
+T = TypeVar("T")
+
+
+class YieldIterator(Exception):
+    """Indicates that a local iterator has no value available.
+
+    This is used internally to implement the union() of multiple blocking
+    local generators."""
+    pass
```

Comment:
Given this seems to be an exception rather than an iterator, it'd be better to suffix it with `Error`.